### PR TITLE
Add new Operations

### PR DESCRIPTION
**Pull request by Robertorosmaninho**
Add support for Operations: Neg, Sub, Div, Rem
Add support for bool literal
Add tests for the new added operations

### CODE REVIEW

File: llvm-examples/sparse_sum_float_test.cc [hunk 1]
```diff
@@ -6,22 +6,22 @@ using namespace taco;
 
 int main() {
     // Create formats
-    Format list({Sparse});
+    Format list({Dense});
```
Comment (guilhermeleobas) [inline llvm-examples/sparse_sum_float_test.cc]: I guess this should be `sparse`
```suggestion
    Format list({Sparse});
```

File: src/codegen/codegen_llvm.cpp [hunk 2]
```diff
@@ -1060,7 +1098,7 @@ void CodeGen_LLVM::visit(const GetProperty* op) {
     case TensorProperty::Indices: {
       auto* mode = get_int_constant(32, op->mode, *this->Context);
       auto* index = get_int_constant(32, op->index, *this->Context);
-
+
```
Comment (guilhermeleobas) [inline src/codegen/codegen_llvm.cpp]:
```suggestion

```

File: llvm-examples/sparse_div_test.cc [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ 
+  //Create tensors
+  Tensor<int32_t> A({3}, list);
+  Tensor<int32_t> B({3}, list);
+  Tensor<int32_t> C({3}, list);
+
+  // Insert data into B
+  B.insert({0}, (int32_t) 4);
+  B.insert({1}, (int32_t) 5);
+  B.insert({2}, (int32_t) 6);
```
Comment (guilhermeleobas) [inline llvm-examples/sparse_div_test.cc]:
```suggestion
  B.insert({0}, (int32_t) 4);
  B.insert({1}, (int32_t) 6);
  B.insert({2}, (int32_t) 8);
```

File: llvm-examples/sparse_div_test.cc [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  // Insert data into C
+  C.insert({0}, (int32_t) 1);
+  C.insert({1}, (int32_t) 2);
+  C.insert({2}, (int32_t) 3);
```
Comment (guilhermeleobas) [inline llvm-examples/sparse_div_test.cc]:
```suggestion
  C.insert({0}, (int32_t) 2);
  C.insert({1}, (int32_t) 3);
  C.insert({2}, (int32_t) 4);
```

File: llvm-examples/sparse_mul_test.cc [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  B.insert({0}, (int32_t) 1);
+  B.insert({1}, (int32_t) 2);
+  B.insert({2}, (int32_t) 3);
+
+  // Insert data into C
+  C.insert({0}, (int32_t) 4);
+  C.insert({1}, (int32_t) 5);
+  C.insert({2}, (int32_t) 6);
+
+  // Insert data into C
```
Comment (guilhermeleobas) [inline llvm-examples/sparse_mul_test.cc]:
```suggestion
  // Insert data into D
```

File: llvm-examples/dense_div_test.cc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
+  // Create tensors
+  Tensor<int32_t> A({3}, list);
+  Tensor<int32_t> B({3}, list);
+  Tensor<int32_t> C({3}, list);
+
+  // Insert data into B
+  B.insert({0}, (int32_t) 4);
+  B.insert({1}, (int32_t) 5);
+  B.insert({2}, (int32_t) 6);
```
Comment (guilhermeleobas) [inline llvm-examples/dense_div_test.cc]:
```suggestion
  B.insert({0}, (int32_t) 4);
  B.insert({1}, (int32_t) 6);
  B.insert({2}, (int32_t) 8);
```

File: llvm-examples/dense_div_test.cc [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+
+  C.insert({0}, (int32_t) 1);
+  C.insert({1}, (int32_t) 2);
+  C.insert({2}, (int32_t) 3);
```
Comment (guilhermeleobas) [inline llvm-examples/dense_div_test.cc]: Let's keep the division in the integer domain
```suggestion
  C.insert({0}, (int32_t) 2);
  C.insert({1}, (int32_t) 3);
  C.insert({2}, (int32_t) 4);
```

**Review by guilhermeleobas — CHANGES_REQUESTED** [review]
Hi @Robertorosmaninho,

Thanks for working on this. I have a few nits to comment, but otherwise the PR looks great. Good job!

File: llvm-examples/sparse_sum_float_test.cc (hunk 1 above)
Comment (Robertorosmaninho) [reply llvm-examples/sparse_sum_float_test.cc]: OK!

File: src/codegen/codegen_llvm.cpp (hunk 2 above)
Comment (Robertorosmaninho) [reply src/codegen/codegen_llvm.cpp]: ok!

File: llvm-examples/sparse_div_test.cc (hunk 3 above)
Comment (Robertorosmaninho) [reply llvm-examples/sparse_div_test.cc]: ok!

File: llvm-examples/sparse_div_test.cc (hunk 4 above)
Comment (Robertorosmaninho) [reply llvm-examples/sparse_div_test.cc]: ok!

File: llvm-examples/dense_div_test.cc (hunk 6 above)
Comment (Robertorosmaninho) [reply llvm-examples/dense_div_test.cc]: ok!

File: llvm-examples/dense_div_test.cc (hunk 7 above)
Comment (Robertorosmaninho) [reply llvm-examples/dense_div_test.cc]: ok!

**Comment by Robertorosmaninho** [comment]
Hey @guilhermeleobas, I fixed the values that you pointed out! The PR should be ready now!
It has been a pleasure to contribute to this project!

**Review by guilhermeleobas — APPROVED** [review]

**Comment by guilhermeleobas** [comment]
Thanks, @Robertorosmaninho

**Pull request merged**